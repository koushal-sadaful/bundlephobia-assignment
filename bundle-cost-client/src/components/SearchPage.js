import React from 'react'
import {push} from 'connected-react-router'
import {bindActionCreators} from 'redux'
import {connect} from 'react-redux'
import {
    fetchSearchSuggestions
} from './../actions/searchActions'
import {Container, Header, Search, Grid, Select} from "semantic-ui-react";
import {updateSearchInputValue} from "../actions/searchActions";
import _ from 'lodash'
import {resetState} from "../actions/packageStatsActions";


class SearchPage extends React.Component {

    searchInputUpdated = (e, {value}) => {
        this.props.updateSearchInputValue(value);
        this.props.fetchSearchSuggestions(value)
    };

    handleResultSelect = (e, {result}) => {
        this.props.updateSearchInputValue(result.title);
        this.props.history.push(`/packageStats/${result.title}`)
    };

    componentWillUnmount() {
        this.props.resetState()
    }

    render() {
        return (
            <div>
                <Container fluid>
                    <Container textAlign='center'>
                        <Header
                            as='h2'
                            content="Fake BundlePhobia "
                            style={{
                                fontSize: '1.2em',
                                marginTop: '2em',
                            }}
                        />
                        <Header
                            as='h5'
                            style={{
                                fontSize: '1.2em',
                                fontWeight: 'normal',
                                marginBottom: 0,
                            }}
                        >
                            <Search
                                fluid
                                className="search-input"
                                value={this.props.searchInputValue}
                                onResultSelect={this.handleResultSelect}
                                onSearchChange={_.debounce(this.searchInputUpdated, 500, {leading: true})}
                                results={this.props.suggestions}
                                placeholder='Type package name'/>

                        </Header>
                    </Container>

                </Container>
            </div>
        )
    }
}


const mapStateToProps = ({search}) => ({
    suggestions: search.suggestions,
    searchInputValue: search.searchInputValue,
});

const mapDispatchToProps = dispatch => bindActionCreators({
    fetchSearchSuggestions,
    updateSearchInputValue,
    resetState,
}, dispatch);

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(SearchPage)