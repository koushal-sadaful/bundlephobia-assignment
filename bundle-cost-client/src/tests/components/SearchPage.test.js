import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {configure, mount} from 'enzyme';
import {Header, Search} from "semantic-ui-react";
import SearchPage from "../../components/SearchPage";
import {createTestStore} from "../testStoreFactory";
import axios from "axios";
import {runTheSnapshotTest} from "../testUtils/TestHelpers";

configure({adapter: new Adapter()});
jest.mock('axios');
jest.useFakeTimers()

describe("SearchPage", () => {
    let wrapper = null;
    afterEach(() => {
        if (wrapper){
            wrapper.unmount()
            wrapper = null
        }

    });

    it("should render a clean search page", () => {
        wrapper = mount(<SearchPage store={createTestStore()}/>);

        const pageTitle = wrapper.find(Header).first().props().content;
        expect(pageTitle).toMatch("Fake BundlePhobia");

        const searchBarProps = wrapper.find(Search).props();
        expect(searchBarProps.placeholder).toMatch("Type package name");

    });

    it("should pass the snapshot test", () => {
        runTheSnapshotTest(<SearchPage store={createTestStore()}/>)
    });
});