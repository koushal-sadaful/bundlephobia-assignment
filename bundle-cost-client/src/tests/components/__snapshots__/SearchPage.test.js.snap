// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage should pass the snapshot test 1`] = `
<div>
  <div
    className="ui fluid container"
  >
    <div
      className="ui center aligned container"
    >
      <h2
        className="ui header"
        style={
          Object {
            "fontSize": "1.2em",
            "marginTop": "2em",
          }
        }
      >
        Fake BundlePhobia 
      </h2>
      <h5
        className="ui header"
        style={
          Object {
            "fontSize": "1.2em",
            "fontWeight": "normal",
            "marginBottom": 0,
          }
        }
      >
        <div
          className="ui fluid search search-input"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
        >
          <div
            className="ui icon input"
          >
            <input
              autoComplete="off"
              className="prompt"
              onChange={[Function]}
              onClick={[Function]}
              placeholder="Type package name"
              tabIndex="0"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="search icon"
              onClick={[Function]}
            />
          </div>
          <div
            className="results transition"
          >
            <div
              className="message empty"
            >
              <div
                className="header"
              >
                No results found.
              </div>
            </div>
          </div>
        </div>
      </h5>
    </div>
  </div>
</div>
`;
